### PR TITLE
Fix admins [PD-1917][PD-1920]

### PR DESCRIPTION
**Pull request by tdphillips**
Improves redirect urls - no longer one urls file with everything dumped into it
Modifies SEO's admin tests to work on all defined apps
Adds redirect middleware to secure and www, adding Destination Manipulation support on those platforms
Adds FSM urls to redirect, adding Company and BUID (and likely others) support to that platform

All tests pass.

### CODE REVIEW

File: seo/tests/setup.py [hunk 1]
```diff
@@ -11,10 +9,12 @@
 from django.test.client import Client
 from django.template import context
 
-from seo_pysolr import Solr
 from import_jobs import DATA_DIR
+from saved_search.groupsearch import SolrGrpEngine, SolrGroupSearchBackend
+from seo.search_backend import DESolrSearchBackend, DESolrEngine
 from seo.tests.factories import BusinessUnitFactory
 from seo.tests.factories import SeoSiteFactory, ConfigurationFactory
+from seo_pysolr import Solr
```
Comment (tdphillips) [inline seo/tests/setup.py]: These import modifications do nothing. I was originally adding my admin tests to [redirect|myjobs|seo]/setup.py, which prompted me to make the imports resemble our style guide.

File: universal/tests/test_admins.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+from seo.tests.setup import DirectSEOBase
+from redirect.tests.setup import RedirectBase
+
+
+def admin_test(self):
+    """
+    Tests add and list views for all models registered with the Django
+    admin.
+    """
+    self.password = 'imbatmancalifornia'
```
Comment (tdphillips) [inline universal/tests/test_admins.py]: Shamelessly stolen from the original admin tests.

File: universal/tests/test_admins.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    self.assertTrue(len(urls) > 0)
+    for url in urls:
+        response = self.client.get(url)
+        # Not all entries (currently postajob/purchasedproduct) have add pages.
+        # Those that don't have add pages return 403 status codes.
+        self.assertIn(response.status_code, [200, 403])
+
+
+class SEOAdminTests(DirectSEOBase):
+    test_admin_add_and_list = admin_test
```
Comment (tdphillips) [inline universal/tests/test_admins.py]: Originally these were `class FooAdminTests(FooBase): pass` and tests were associated with the classes as follows:
```
for class_ in [SEOAdminTests, ...]:
    setattr(class_, 'test_admin_add_and_list', admin_test)
```

But that caused the redirect version to get ran twice.

File: redirect_settings.py [hunk 4]
```diff
@@ -24,10 +24,6 @@
     }
 }
 
-EXCLUDED_VIEW_SOURCE_CACHE_KEY = 'excluded_view_sources'
-
-CUSTOM_EXCLUSION_CACHE_KEY = 'custom_excluded_view_sources'
-
 MIDDLEWARE_CLASSES = global_settings.MIDDLEWARE_CLASSES + (
     'redirect.middleware.ExcludedViewSourceMiddleware',
```
Comment (tdphillips) [inline redirect_settings.py]: I don't much like leaving this here but, given we just recently made redirect ignore middleware from other classes, it had to be done.

**Comment by koonsies** [comment]
No Xenomorphs found :+1:

**Pull request merged**